ENH: Add logo_link theme option (#246)
pandas-dev/pydata-sphinx-theme: docs/user_guide/configuring.rst

   html_logo = "_static/logo.png"

> The logo links to ``master_doc`` by default. If you'd like to link to other page

Reviewer: ```suggestion
The logo links to the first page of your documentation by default. If you'd like to link to another page
```

Author: Agree on "other page" to "another page" but the logo really links to ``master_doc`` by default though and sometimes ``master_doc`` is not the landing page (e.g. numpy dev docs).

Reviewer: hmmm - fair enough, what do you think about the new language suggestion? The thing is that most users don't know what `master_doc` is so I prefer using terminology that the "average" user will likely understand

Reviewer: another thing we could do instead is include a link to the Sphinx documentation for `master_doc`

Author: I think the new language suggestion is pretty informative!

If you'd like it to link to another page or use an external link instead, use the following configuration:

.. code:: python

   html_theme_options = {

>       "logo_link": "<other page or external link>"
> 

Reviewer 2: ```suggestion
       "logo_link": "<other page or external link>"
   }

```
